src/background/tokenBackground.ts

```diff
@@ -118,6 +180,28 @@ export default class TokenBackground {
     }
   }
 
+  private addToken = async (contractAddress: string, name: string, symbol: string, decimals: number) => {
+    const newToken = new QRCToken(name, symbol, decimals, contractAddress);
+    this.tokens!.push(newToken);
+    this.setTokenListInChromeStorage();
+    await this.getQRCTokenBalance(newToken);
+  }
+
+  private removeTokenAtIndex = (index: number) => {
```

Comment:
lets find and remove the token by `contractAddress` instead of `index`.

Reply:
I considered doing that - passing the contractAddress rather than the index from the UI, as it is possibly a little clearer at the background level which token is getting removed, but it is also extra computation, because removing by contractAddress means we have to use the findIndex method to iterate through the array to determine the index to remove the token from the array, and we already have the index. Do you think it is worth it?

Reply:
Ya it's clearer to remove by a definitive property and less chance of a bug to pop up from this.